cmake/init.cmake

```diff
@@ -1,51 +1,43 @@
 # Distributed under the OpenDDS License. See accompanying LICENSE
 # file or http://www.opendds.org/license.html for details.
 
-include(${CMAKE_CURRENT_LIST_DIR}/config.cmake)
-
-if(NOT DEFINED OPENDDS_DEBUG)
-  set(OPENDDS_DEBUG ON)
-endif()
-
-if(NOT DEFINED OPENDDS_INLINE)
-  set(OPENDDS_INLINE ON)
-endif()
-
-if(NOT DEFINED OPENDDS_BUILT_IN_TOPICS)
-  set(OPENDDS_BUILT_IN_TOPICS ON)
-endif()
-
-if(NOT DEFINED OPENDDS_OBJECT_MODEL_PROFILE)
-  set(OPENDDS_OBJECT_MODEL_PROFILE ON)
-endif()
-
-if(NOT DEFINED OPENDDS_PERSISTENCE_PROFILE)
-  set(OPENDDS_PERSISTENCE_PROFILE ON)
+if(_OPENDDS_INIT_CMAKE)
+  return()
 endif()
+set(_OPENDDS_INIT_CMAKE TRUE)
 
-if(NOT DEFINED OPENDDS_OWNERSHIP_PROFILE)
-  set(OPENDDS_OWNERSHIP_PROFILE ON)
-endif()
-if(NOT DEFINED OPENDDS_OWNERSHIP_KIND_EXCLUSIVE)
-  set(OPENDDS_OWNERSHIP_KIND_EXCLUSIVE ${OPENDDS_OWNERSHIP_PROFILE})
-endif()
+find_package(Perl REQUIRED)
```

Comment:
For now yes that's all. It should be noted it was required before, [it was just doing it a different way](https://github.com/OpenDDS/OpenDDS/blob/27ea3728477342c43e71a618cf6476cf7c3e6728/cmake/OpenDDSConfig.cmake#L388-L395). The reason it's here is there's a Perl script to get ACE's features is right after this in my local changes and I didn't want it to find Perl more than once. We could theoretically make it optional, but there is a case to be made that it could be required:

1. The user should have Perl if they built ACE/TAO.
2. If they didn't build ACE/TAO, we will probably need Perl to build ACE/TAO at some point.
3. If we need to do other mostly isolated things like investigate or build ACE, it's a lot easier those things in Perl than CMake.
4. It lets us reuse code we've already written.

I'm not committed to this yet though because obviously if we could completely eliminate depending on Perl in the future, that would simplify things for the user.

Reply:
Assuming `generate_export_file.pl` was eliminated, does it make sense to have Perl be a conditional dependency in CMake?  So if the control flow reaches this `find_package(Perl)` then it really is required, but not all configurations would get there?

Reply:
As discussed, I will have `tao_idl` generate it probably, though it can also be made by `configure_file`. I would like to defer that for the next PR where I'm planning on doing more with `opendds_target_sources`. I can make Perl optional there or remove it depending on how much we want to rewrite in CMake.